### DISCUSSION TITLE
Feedback for sensitive boards cargo ordering

### DISCUSSION THREAD

**Discussion by StrangeWeirdKitten** [Bubberstation/Bubberstation#724] [codebase discussion]
Feedback for #719

**Comment by YakumoChen** [comment]
The AI should not be overly difficult to subvert. 

The AI has a large impact on antagonists regardless of its lawset (and heavily based on who is playing the AI). While it should be treated as an antag-neutral role it often is not. 

The AI should not be a reliable ally to the crew, it is extremely powerful and influential on the round, which the crew should be paranoid about its allegiance at any given time.

If the AI is too bothersome to antagonists and subverting is too easy to gate or deny, antags may just kill the AI instead and remove it from the round.

**Comment by YakumoChen** [comment]
Being subverted is also a large part of the AI experience and this will simply be a negative for many AI players who enjoy it

**Comment by SovietSpaghetti** [comment]
It kinda feels like the AI either gets subverted or is malf nearly every other round. It should be a big deal when an entire department of the station, specifically, the department that has access to every single system on it and access to some of the most powerful combat tools in the game (engi borg stun baton, peacekeeper airhorn and hug module, med piercing injector), is wordlessly converted to an antag because they managed to hack through the one door through maint stopping them from getting to a circuit printer.

**Comment by SovietSpaghetti** [comment]
I feel like this is a good change, silicons as a department are strong as shit, especially when malf or hacked. They're the only department in the game that gets all access, a private comms system no-one else can access without a traitor item that even works when comms are down, they get every language in the game, every comms channel, and access to all machines at range. They're _also_ the only department that can be turned entirely to an antag's side with the use of one circuit board in maints and a cheap item from an uplink.

**Comment by StrangeWeirdKitten** [comment]
It's easy to subvert. It's easy to relaw on both sides. When it happens, it's just law ping pong with silicons stuck in the middle. Sure, getting subverted is fun. But a constant battle between two people with uploads is not. There are still multiple ways for you to subvert an AI/Borgs as robotics. You can manually use the law modules themselves. Typically when situations occur with random maint uploads, they're practically untracable. The AI gets 33 law changes and the emagged borgs start looking to activate their self-destruct after the 2nd or 3rd time they were shot by security. If an AI wanted to stop people from tampering with its laws, then it would have to deconstruct two critical lathes. (and this has happened multiple times)

The order is a supply pack. You can just order it inside the science request console. I'm not worried about this making it harder for robotics to subvert silicons. Because it's already been extremely easy, and this makes practically zero difference when most silicon subversions from robotics are with an emag.

**Comment by YakumoChen** [comment]
Make the boards themselves tracable on GPS and you solve 90% of the ping-pong issue you bring up

I have severe doubt it will be easy to change AI laws ordering a cargo crate that practically screams "AI Subversion Crate"

Many boards are also CE exclusive, not RD. Why does only science get this crate?

**Comment by YakumoChen** [comment]
> If an AI wanted to stop people from tampering with its laws, then it would have to deconstruct two critical lathes. (and this has happened multiple times)

All this really adds to this problem is one more lathe and a couple consoles it has to destroy, making this gated to cargo just invites the AI to destroy cargo, and then destroy all the printers it had to destroy before that can print any board.

**Comment by StrangeWeirdKitten** [comment]
I can change the access requirements. 

> I have severe doubt it will be easy to change AI laws ordering a cargo crate that practically screams "AI Subversion Crate"

As I said, there are multiple ways to subvert an AI. It's an extremely powerful tool, and there should be at least a little risk involved. 

> All this really adds to this problem is one more lathe and a couple consoles it has to destroy, making this gated to cargo just invites the AI to destroy cargo, and then destroy all the printers it had to destroy before that can print any board.

PDAs have NT IRN and science typically has a 300,000 budget. Also, if you do this as an AI. You get your core raided. At least you can manually upload new laws directly into an AI core.

**Comment by StrangeWeirdKitten** [comment]
At the end of the day, it's difficult to speculate what gameplay this will result in. Hence, I will be watching it. It's a test merge and the idea is still under review.

**Comment by YakumoChen** [comment]
> As I said, there are multiple ways to subvert an AI. It's an extremely powerful tool, and there should be at least a little risk involved.

The main ways are:
Secure Tech Storage, which is very easy to gate antags from looting
Buying a crate that betrays extremely obvious intentions if you're not a head of staff 
Literally breaking into the AI core to tap a law board on it
Building a new AI, requires a willing posi-brain (never happens)

Aside from STC, most of these are wildly unfeasible

The board can't even be ordered unless you use a dept console, work in cargo, or have Command access to open the crate in the first place.

**Comment by YakumoChen** [comment]
I would like an answer why you would rather not consider simply making the boards more tracable (via GPS) to remove the unfindable upload problem

**Comment by YakumoChen** [comment]
> PDAs have NT IRN and science typically has a 300,000 budget. Also, if you do this as an AI. You get your core raided. At least you can manually upload new laws directly into an AI core.

Just destroy the NT net relay in tcomms and you remove this issue

The AI gets its core raided before, so I don't know what this changes
